python/pyspark/sql/session.py

```diff
@@ -1410,28 +1410,35 @@ def prepare(obj: Any) -> Any:
         df._schema = struct
         return df
 
-    def sql(self, sqlQuery: str, args: Optional[Dict[str, Any]] = None, **kwargs: Any) -> DataFrame:
+    def sql(
+        self, sqlQuery: str, args: Optional[Union[Dict[str, Any], List]] = None, **kwargs: Any
+    ) -> DataFrame:
         """Returns a :class:`DataFrame` representing the result of the given query.
         When ``kwargs`` is specified, this method formats the given string by using the Python
-        standard formatter. The method binds named parameters to SQL literals from `args`.
+        standard formatter. The method binds named parameters to SQL literals or
+        positional parameters from `args`. It doesn't support named and positional parameters
+        in the same SQL query.
 
         .. versionadded:: 2.0.0
 
         .. versionchanged:: 3.4.0
             Supports Spark Connect and parameterized SQL.
 
+        .. versionchanged:: 3.5.0
+            Supports parameterized SQL with positional parameters.
```

Comment:
nit: No need for "Supports" since this section is all about what has been changed (added?)

Reply:
ok